scalafmt-core/shared/src/main/scala/org/scalafmt/util/TreeOps.scala

```diff
@@ -676,7 +679,14 @@ object TreeOps {
     }
 
   // Redundant {} block around case statements
-  def isCaseBodyABlock(ft: FormatToken, caseStat: Case): Boolean =
-    ft.right.is[Token.LeftBrace] && (caseStat.body eq ft.meta.rightOwner)
+  def isCaseBodyABlock(ft: FormatToken, caseStat: Tree): Boolean = {
+    val body = caseStat match {
+      case tc: TypeCase => Some(tc.body)
+      case c: Case => Some(c.body)
+      case _ => None
+    }
+    body.isDefined && ft.right
```

Comment:
same comment as above, with using exists or null.

yet another option, check left brace first, then body:
```
ft.right.is[...] && ft.meta.rightOwner.eq(
  caseStat match {
    case x: Case => x.body
    ...
    case _ => null
 }
)

Reply:
Added exists here and above, good idea!